src/Uno.UI.Runtime.Skia.X11/X11XamlRootHost.cs

```diff
@@ -46,33 +47,75 @@ internal partial class X11XamlRootHost : IXamlRootHost
 
 	public X11Window X11Window => _x11Window!.Value;
 
-	public X11XamlRootHost(Window winUIWindow, Action<Size> resizeCallback, Action closeCallback, Action<bool> focusCallback, Action<bool> visibilityCallback)
+	public X11XamlRootHost(Window winUIWindow, Action<Size> resizeCallback, Action closingCallback, Action<bool> focusCallback, Action<bool> visibilityCallback)
 	{
 		_window = winUIWindow;
 
 		_resizeCallback = resizeCallback;
-		_closeCallback = closeCallback;
+		_closingCallback = closingCallback;
 		_focusCallback = focusCallback;
 		_visibilityCallback = visibilityCallback;
 
 		_applicationView = ApplicationView.GetForWindowId(winUIWindow.AppWindow.Id);
 		_applicationView.PropertyChanged += OnApplicationViewPropertyChanged;
+		var coreApplicationView = CoreApplication.GetCurrentView();
+		coreApplicationView.TitleBar.ExtendViewIntoTitleBarChanged += UpdateWindowPropertiesFromCoreApplication;
 
 		_closed = new TaskCompletionSource();
 		Closed = _closed.Task;
-		Closed.ContinueWith(_ => _applicationView.PropertyChanged -= OnApplicationViewPropertyChanged);
+		Closed.ContinueWith(_ =>
+		{
+			_applicationView.PropertyChanged -= OnApplicationViewPropertyChanged;
+			coreApplicationView.TitleBar.ExtendViewIntoTitleBarChanged -= UpdateWindowPropertiesFromCoreApplication;
+		});
```

Comment:
Thinking about it, this may not be executed on the appropriate thread (this is a background there in this context). We may need to schedule on the dispatcher.

Reply:
Unsubscribing should be thread-safe, so why would we need to schedule this on the dispatcher thread?

Reply:
I think it is ok in these cases, especially as these APIs are UWP-based anyway